router/src/db.rs

```diff
@@ -135,13 +129,19 @@ impl Db {
                     return None;
                 }
             }
+            // Batch size
+            let size = requests.len();
+
+            let mut entries = HashMap::with_capacity(size);
```

Comment:
`IntMap` from the [nohash_hasher](https://docs.rs/nohash-hasher/latest/nohash_hasher/) crate is much faster for integer keys. The default HashMap hashing is kind of slow.

Reply:
How much faster are we talking about here?

Reply:
Everything is relative :) but it's a drop-in replacement. Here is a [reference](https://nnethercote.github.io/perf-book/hashing.html).

Reply:
done